actionpack/test/dispatch/uri_test.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "abstract_unit"
+
+class URITest < ActiveSupport::TestCase
+  test "create new url" do
+    uri = ActionDispatch::Http::URI.new("http://myapp.test/page?id#me")
+    assert_equal uri.to_s, "http://myapp.test/page?id#me"
+    assert_equal uri.scheme, "http"
+    assert_equal uri.protocol, "http://"
+    assert_equal uri.host, "myapp.test"
+    assert_equal uri.path, "/page"
+    assert_equal uri.fragment, "me"
+  end
+
+  test "create with ipv6" do
+    uri = ActionDispatch::Http::URI.new("http://2001:0db8:85a3:0000:0000:8a2e:0370:7334")
```

Comment:
Ah, that answers my earlier question, but opens a new one: where do we expect to see such an invalid URL, and why do we care that we can parse it?

Reply:
it is from the tests of the current url module

Reply:
I can't find this "URI" string anywhere at the moment... got a link?

Reply:
https://github.com/rails/rails/blob/main/actionpack/test/dispatch/request_test.rb#L146

Reply:
Ah okay, yeah, so the input URL to produce that env should be `http://[2001:0db8:85a3:0000:0000:8a2e:0370:7334]`, the same as the `to_s` output in this assertion. It's not valid to put an IPv6 literal into a URL without the `[]` delimiters (which is why `URI.parse` isn't accepting it).

Reply:
that is true, and it would be possible to remove the handling of incorrect IPv6 urls. I added them as inside the test of actionpack those incorrect urls are used and in case you want to create a URI for each request, this use case would need to be handled. For the moment, as this part got removed, we could also remove the handling of those not 100% correct urls. ( It seems they exist in the real world, or why do they got added in many of the test cases? )